# Avaliação 1
#### Valor : 50 pontos (cada questão vale 2 pontos)

GRUPO:





Considere a seguinte lista para resolver as questões de 1 a 4:

In [1]:
lista = [2, 29, 8, 98, 10, 3, -4, 9, 10]



1) Imprima a soma do maior e o menor elemento da lista:

In [2]:
print(min(lista) + max(lista))

94


2) Imprima o número de ocorrências do maior elemento da lista:

In [3]:
print(lista.count(max(lista)))

1


3) Imprima o elemento no meio da lista:

In [4]:
print(len(lista))
print(lista[int(round(len(lista) / 2, 0))])

9
10


4) Imprima o maior elemento da lista elevado ao quadrado:

In [5]:
print(min(lista) ** 2)

16


5) Crie uma lista com todos os números pares de 0 até 1000.

In [6]:
lista_par = []
for n in range(0, 101):
    if(n % 2 == 0):
        lista_par.append(n)
print(lista_par)

[0, 2, 4, 6, 8, 10, 12, 14, 16, 18, 20, 22, 24, 26, 28, 30, 32, 34, 36, 38, 40, 42, 44, 46, 48, 50, 52, 54, 56, 58, 60, 62, 64, 66, 68, 70, 72, 74, 76, 78, 80, 82, 84, 86, 88, 90, 92, 94, 96, 98, 100]


6) Crie uma lista com todos os números ímpares de 0 até 10.000.

In [7]:
lista_impar = []
for n in range(0, 10001):
    if(n % 2 != 0):
        lista_impar.append(n)
print(lista_impar)

[1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 39, 41, 43, 45, 47, 49, 51, 53, 55, 57, 59, 61, 63, 65, 67, 69, 71, 73, 75, 77, 79, 81, 83, 85, 87, 89, 91, 93, 95, 97, 99, 101, 103, 105, 107, 109, 111, 113, 115, 117, 119, 121, 123, 125, 127, 129, 131, 133, 135, 137, 139, 141, 143, 145, 147, 149, 151, 153, 155, 157, 159, 161, 163, 165, 167, 169, 171, 173, 175, 177, 179, 181, 183, 185, 187, 189, 191, 193, 195, 197, 199, 201, 203, 205, 207, 209, 211, 213, 215, 217, 219, 221, 223, 225, 227, 229, 231, 233, 235, 237, 239, 241, 243, 245, 247, 249, 251, 253, 255, 257, 259, 261, 263, 265, 267, 269, 271, 273, 275, 277, 279, 281, 283, 285, 287, 289, 291, 293, 295, 297, 299, 301, 303, 305, 307, 309, 311, 313, 315, 317, 319, 321, 323, 325, 327, 329, 331, 333, 335, 337, 339, 341, 343, 345, 347, 349, 351, 353, 355, 357, 359, 361, 363, 365, 367, 369, 371, 373, 375, 377, 379, 381, 383, 385, 387, 389, 391, 393, 395, 397, 399, 401, 403, 405, 407, 409, 411, 413, 415, 417, 419, 421,

7) Imprima o valor da média das listas dos items 5 e 6.

In [8]:
for n in lista_par:
    print(n / 2)

0.0
1.0
2.0
3.0
4.0
5.0
6.0
7.0
8.0
9.0
10.0
11.0
12.0
13.0
14.0
15.0
16.0
17.0
18.0
19.0
20.0
21.0
22.0
23.0
24.0
25.0
26.0
27.0
28.0
29.0
30.0
31.0
32.0
33.0
34.0
35.0
36.0
37.0
38.0
39.0
40.0
41.0
42.0
43.0
44.0
45.0
46.0
47.0
48.0
49.0
50.0


In [9]:
for n in lista_impar:
    print(n / 2)

0.5
1.5
2.5
3.5
4.5
5.5
6.5
7.5
8.5
9.5
10.5
11.5
12.5
13.5
14.5
15.5
16.5
17.5
18.5
19.5
20.5
21.5
22.5
23.5
24.5
25.5
26.5
27.5
28.5
29.5
30.5
31.5
32.5
33.5
34.5
35.5
36.5
37.5
38.5
39.5
40.5
41.5
42.5
43.5
44.5
45.5
46.5
47.5
48.5
49.5
50.5
51.5
52.5
53.5
54.5
55.5
56.5
57.5
58.5
59.5
60.5
61.5
62.5
63.5
64.5
65.5
66.5
67.5
68.5
69.5
70.5
71.5
72.5
73.5
74.5
75.5
76.5
77.5
78.5
79.5
80.5
81.5
82.5
83.5
84.5
85.5
86.5
87.5
88.5
89.5
90.5
91.5
92.5
93.5
94.5
95.5
96.5
97.5
98.5
99.5
100.5
101.5
102.5
103.5
104.5
105.5
106.5
107.5
108.5
109.5
110.5
111.5
112.5
113.5
114.5
115.5
116.5
117.5
118.5
119.5
120.5
121.5
122.5
123.5
124.5
125.5
126.5
127.5
128.5
129.5
130.5
131.5
132.5
133.5
134.5
135.5
136.5
137.5
138.5
139.5
140.5
141.5
142.5
143.5
144.5
145.5
146.5
147.5
148.5
149.5
150.5
151.5
152.5
153.5
154.5
155.5
156.5
157.5
158.5
159.5
160.5
161.5
162.5
163.5
164.5
165.5
166.5
167.5
168.5
169.5
170.5
171.5
172.5
173.5
174.5
175.5
176.5
177.5
178.5
179.5
180.5
181.5
182.5
183.5
184.5


1591.5
1592.5
1593.5
1594.5
1595.5
1596.5
1597.5
1598.5
1599.5
1600.5
1601.5
1602.5
1603.5
1604.5
1605.5
1606.5
1607.5
1608.5
1609.5
1610.5
1611.5
1612.5
1613.5
1614.5
1615.5
1616.5
1617.5
1618.5
1619.5
1620.5
1621.5
1622.5
1623.5
1624.5
1625.5
1626.5
1627.5
1628.5
1629.5
1630.5
1631.5
1632.5
1633.5
1634.5
1635.5
1636.5
1637.5
1638.5
1639.5
1640.5
1641.5
1642.5
1643.5
1644.5
1645.5
1646.5
1647.5
1648.5
1649.5
1650.5
1651.5
1652.5
1653.5
1654.5
1655.5
1656.5
1657.5
1658.5
1659.5
1660.5
1661.5
1662.5
1663.5
1664.5
1665.5
1666.5
1667.5
1668.5
1669.5
1670.5
1671.5
1672.5
1673.5
1674.5
1675.5
1676.5
1677.5
1678.5
1679.5
1680.5
1681.5
1682.5
1683.5
1684.5
1685.5
1686.5
1687.5
1688.5
1689.5
1690.5
1691.5
1692.5
1693.5
1694.5
1695.5
1696.5
1697.5
1698.5
1699.5
1700.5
1701.5
1702.5
1703.5
1704.5
1705.5
1706.5
1707.5
1708.5
1709.5
1710.5
1711.5
1712.5
1713.5
1714.5
1715.5
1716.5
1717.5
1718.5
1719.5
1720.5
1721.5
1722.5
1723.5
1724.5
1725.5
1726.5
1727.5
1728.5
1729.5
1730.5
1731.5
1732.5
1733.5

3202.5
3203.5
3204.5
3205.5
3206.5
3207.5
3208.5
3209.5
3210.5
3211.5
3212.5
3213.5
3214.5
3215.5
3216.5
3217.5
3218.5
3219.5
3220.5
3221.5
3222.5
3223.5
3224.5
3225.5
3226.5
3227.5
3228.5
3229.5
3230.5
3231.5
3232.5
3233.5
3234.5
3235.5
3236.5
3237.5
3238.5
3239.5
3240.5
3241.5
3242.5
3243.5
3244.5
3245.5
3246.5
3247.5
3248.5
3249.5
3250.5
3251.5
3252.5
3253.5
3254.5
3255.5
3256.5
3257.5
3258.5
3259.5
3260.5
3261.5
3262.5
3263.5
3264.5
3265.5
3266.5
3267.5
3268.5
3269.5
3270.5
3271.5
3272.5
3273.5
3274.5
3275.5
3276.5
3277.5
3278.5
3279.5
3280.5
3281.5
3282.5
3283.5
3284.5
3285.5
3286.5
3287.5
3288.5
3289.5
3290.5
3291.5
3292.5
3293.5
3294.5
3295.5
3296.5
3297.5
3298.5
3299.5
3300.5
3301.5
3302.5
3303.5
3304.5
3305.5
3306.5
3307.5
3308.5
3309.5
3310.5
3311.5
3312.5
3313.5
3314.5
3315.5
3316.5
3317.5
3318.5
3319.5
3320.5
3321.5
3322.5
3323.5
3324.5
3325.5
3326.5
3327.5
3328.5
3329.5
3330.5
3331.5
3332.5
3333.5
3334.5
3335.5
3336.5
3337.5
3338.5
3339.5
3340.5
3341.5
3342.5
3343.5
3344.5

4831.5
4832.5
4833.5
4834.5
4835.5
4836.5
4837.5
4838.5
4839.5
4840.5
4841.5
4842.5
4843.5
4844.5
4845.5
4846.5
4847.5
4848.5
4849.5
4850.5
4851.5
4852.5
4853.5
4854.5
4855.5
4856.5
4857.5
4858.5
4859.5
4860.5
4861.5
4862.5
4863.5
4864.5
4865.5
4866.5
4867.5
4868.5
4869.5
4870.5
4871.5
4872.5
4873.5
4874.5
4875.5
4876.5
4877.5
4878.5
4879.5
4880.5
4881.5
4882.5
4883.5
4884.5
4885.5
4886.5
4887.5
4888.5
4889.5
4890.5
4891.5
4892.5
4893.5
4894.5
4895.5
4896.5
4897.5
4898.5
4899.5
4900.5
4901.5
4902.5
4903.5
4904.5
4905.5
4906.5
4907.5
4908.5
4909.5
4910.5
4911.5
4912.5
4913.5
4914.5
4915.5
4916.5
4917.5
4918.5
4919.5
4920.5
4921.5
4922.5
4923.5
4924.5
4925.5
4926.5
4927.5
4928.5
4929.5
4930.5
4931.5
4932.5
4933.5
4934.5
4935.5
4936.5
4937.5
4938.5
4939.5
4940.5
4941.5
4942.5
4943.5
4944.5
4945.5
4946.5
4947.5
4948.5
4949.5
4950.5
4951.5
4952.5
4953.5
4954.5
4955.5
4956.5
4957.5
4958.5
4959.5
4960.5
4961.5
4962.5
4963.5
4964.5
4965.5
4966.5
4967.5
4968.5
4969.5
4970.5
4971.5
4972.5
4973.5

8) Crie 3 tuplas para armazenar as notas dos alunos em uma disciplina. Cada tupla deverá conter o nome e sobrenome do aluno (escolhido por você) seguido de suas duas notas que deverão, por sua vez, estar armazenadas em uma lista.

In [10]:
mateus = ('Mateus Luz', 10.0, 9.5)
willian = ('Willian Dener', 9.0, 8.5)
leonardo = ('Leonardo Chalhoub', 8.0, 7.5)
print(mateus, willian, leonardo)

('Mateus Luz', 10.0, 9.5) ('Willian Dener', 9.0, 8.5) ('Leonardo Chalhoub', 8.0, 7.5)


9) Imprima a soma das notas de cada aluno do exercício 8, use as variáveis criadas por você:

In [11]:
print(f'A soma das notas do aluno', mateus[0], f'é', mateus[1] + mateus[2])
print(f'A soma das notas do aluno', willian[0], f'é', willian[1] + willian[2])
print(f'A soma das notas do aluno', leonardo[0], f'é', leonardo[1] + leonardo[2])

A soma das notas do aluno Mateus Luz é 19.5
A soma das notas do aluno Willian Dener é 17.5
A soma das notas do aluno Leonardo Chalhoub é 15.5


10) Imprima o nome e a maior nota de cada aluno do exercício 8, use as variáveis criadas por você:

In [12]:
print(f'A maior nota do aluno', mateus[0], f'é', max([mateus[1], mateus[2]]))
print(f'A maior nota do aluno', willian[0], f'é', max([willian[1], willian[2]]))
print(f'A maior nota do aluno', leonardo[0], f'é', max([leonardo[1], leonardo[2]]))

A maior nota do aluno Mateus Luz é 10.0
A maior nota do aluno Willian Dener é 9.0
A maior nota do aluno Leonardo Chalhoub é 8.0


11) Construa um dicionário para armazenar as notas de 5 alunos (Os dados são criados por você).

In [13]:
dicionario_notas = {
    "Leonardo": 9.75,
    "Mateus": 8.5,
    "Murilo": 8.75,
    "Nelson": 7.5,
    "Willian": 9.5
}

12) Usando o dicionário criado no exercício 11, imprima o valor da maior nota. A reposta deve ter o seguinte formato:

**A maior nota da turma é XX**, onde XX é o valor encontrado.

In [14]:
print(f'A maior nota da turma é', max(dicionario_notas.values()))

A maior nota da turma é 9.75


13) Sabendo que A = {0, 1, 2, 3, 4, 5, 6}, B = {6, 7, 8, 9} e C = {2, 4, 6, 8, 10}, quais são os elementos do conjunto (A∩B)UC?

In [15]:
A = {0, 1, 2, 3, 4, 5, 6}
B = {6, 7, 8, 9}
C = {2, 4, 6, 8, 10}

In [16]:
def intersec(listaA, listaB):
    listaAB = []
    for x in listaA:
        if x in listaB:
            listaAB.append(x)
    return listaAB

listaAB = intersec(A, B)

def uni(listaB, listaC):
    listaAC = []
    for x in listaB:
        listaAC.append(x)
    for x in listaC:
        listaAC.append(x)
    return listaAC

print(uni(listaAB, C))

[6, 2, 4, 6, 8, 10]


14) Considere as listas abaixo para fazer o exercício:

In [17]:
alunos_natação = ['Aline Soares', 'Bianca Araújo', 'André da Silva', 'Romeu Cunha', 'Luisa Abreu', 'Aline Soares']
alunos_pedal = ['Antonio José', 'Maria Costa', 'Romeu Cunha', 'Luisa Abreu', 'Maria Luisa', 'Jairo Ramos']
alunos_corrida = ['Romeu Cunha', 'Gustavo Andrade', 'Henrique Loureiro', 'Luana Pereira', 'Aline Soares', 'Bruna Fernandes']

14.1) Excluir os nomes duplicados em cada modalidade, ou seja, cada modalidade só poderá conter um nome para cada aluno. 

14.2) Verificar o total de alunos inscrito em
cada modalidade

14.3) Imprimir uma listagem com o nome de todos os alunos (incluindo as 3 modalidades) e a quantidade total de alunos (incluindo as 3 modalidades)

14.4) Imprimir uma listagem para cada combinação de duas modalidades possíveis com o nome de cada aluno inscrito em ambas as modalidades e o seu total

* Listagem 1: Alunos natação e alunos pedal - nomes e quantidade de alunos
* Listagem 2: Alunos natação e alunos corrida - nomes e quantidade de alunos
* Listagem 3: Alunos pedal e alunos corrida -nomes e quantidade de alunos


15) Abaixo é apresentado um dicionário contendo os seguintes dados de alguns vencedores do Prêmio Nobel da Paz:
{Ano da premiação: (Nome, País de Origem, Ano de Nascimento, Motivação do prêmio, gênero (Feminino, Masculino ou Organização))}

In [18]:
nobel = {2011: [['Ellen Jonhson Sirleaf', 'Liberia', 1938, 'woman rights', "F"],['Leymah Gbowee', 'Liberia', 1972, 'woman rights', 'F'], ['Tawakkol', 'Liberia', 1979, 'woman rights', 'F' ]],
         2012: ['Eurpean Union', 'Europe', 1952, 'human rights','O'],
         2013: ['Organization for the Prohibition of Chemical Weapons', 'The Netherlands', 1997, 'eliminate chemical weapons', 'O'],
         2014: [['Kailash Satyarthi', 'India', 1954, 'children education', 'M'], ['Malala Yousafzai', 'Pakistan', 1997, 'children education', 'O']],
         2015: ['National Dialogue Quartet', 'Tunisia', 2011, 'democracy', 'O'],
         2016: ['Juan Manuel Santos', 'Colombia', 1951, 'end of civil war', 'O'],
         2017: ['International Campaign to Abolish Nuclear Weapons', 'Australia', 2007, 'nuclear weapons prohibition', 'O'],
         2018: ['Denis Mukwege', 'Congo', 1955, 'end sexual violation at war', 'M'],
         2019: ['Abiy Ahmed Ali', 'Ethiopia', 1976, 'international conflicts', 'M'],
         2020: ['World Food Programme', 'Italy', 1961, 'combat hunger', 'O']}

15.1) Quais foram os vencedores do Prêmio Nobel da Paz em 2011?

15.2) Imprima a relação dos países de origem dos laureados com o Prêmio Nobel da Paz. A relação deverá conter apenas os nomes dos países.

16) Em uma competição de salto em distância cada atleta tem direito a cinco saltos. O resultado do atleta será determinado pela média dos cinco valores de cada salto. Você deve fazer um programa que receba o nome e as cinco distâncias alcançadas pelo atleta em seus saltos (via input()) e depois informe o nome, os saltos e a média dos saltos. O programa deve ser encerrado quando não for informado o nome do atleta. A saída do programa deve ser conforme o exemplo abaixo:

Atleta: Rodrigo
 
Primeiro Salto: 6.5 m

Segundo Salto: 6.1 m

Terceiro Salto: 6.2 m

Quarto Salto: 5.4 m

Quinto Salto: 5.3 m

Resultado final:

Atleta: Rodrigo

Saltos: 6.5 - 6.1 - 6.2 - 5.4 - 5.3

Média dos saltos: 5.9 m


17) Qual é a diferença entre os loops **while** e **for**?

18) Faça um Programa que peça a idade e a altura de 3 pessoas, armazene cada informação no seu respectivo vetor. Imprima a idade e a altura de cada pessoa.

19) Dados os dois vetores abaixo (listas), faça um programa que leia os dois vetores e gere um terceiro vetor de 20 elementos, cujos valores deverão ser compostos pelos elementos intercalados dos dois outros vetores.

L1 = [2,4,6,8,10,12,14,16,18,20]

L2 = [1,3,5,7,9,11,13,15,17,19]

20) Faça uma função que receba um valor n e que imprima:
  
    1
  
    2   2
  
    3   3   3
  
    .....
  
    n   n   n   n   n   n  ... n

21) Faça uma função que receba um valor n e que imprima:
  
  1
  
  1   2
  
  1   2   3
  
  .....
  
  1   2   3   ...  n

22) Faça uma função que retorne o reverso de um número inteiro passado por parâmetro. Por exemplo: 127 -> 721.

23) Faça uma função que receba dois vetores com 10 elementos cada. Sua função deverá retornar um terceiro vetor de 20 elementos, cujos valores deverão ser compostos pelos elementos intercalados dos dois outros vetores. OBS: Apesar de parecido, este exercício não é idêntico ao 19. Obs: No exercício anterior, o código não é uma função.

24) Faça uma função que receba uma string e retorne a sua quantidade de consoantes.

25) Altere o exercício 25 para que o retorno seja um dicionário, onde a chaves são as consoantes e os valores, a quantidade de vezes que a consoante aparece na string